src/app/issue/settings/settings.component.ts

```diff
@@ -2,133 +2,213 @@ import { Component, OnInit } from '@angular/core';
 })
 export class SettingsComponent implements OnInit {
-  name: string;
-  description: string;
-  priority: number;
-  type: string;
-  state: string;
-  visibleInput: string;
-  visible: boolean;
-  startDate: Date;
-  endDate: Date;
+  visibleInput: string = '';
+  stateActive: boolean = true;
+  companyId: string;
+  projectId: string;
+  issueId: string;
 
   constructor(
     private router: Router,
     private issueService: IssueService,
-    private route: ActivatedRoute
+    private issueDetailsService: IssueDetailsService,
+    private issueAssignedUsersService: IssueAssignedUsersService,
+    private issuePredecessorService: IssuePredecessorService,
+    private route: ActivatedRoute,
+    private projectService: ProjectService,
+    private modal: NzModalService
   ) {}
 
-  companyId: string;
-  projectId: string;
   ngOnInit(): void {
     this.companyId = this.route.snapshot.paramMap.get('companyId');
     this.projectId = this.route.snapshot.paramMap.get('projectId');
+    this.issueId = this.route.snapshot.paramMap.get('issueId');
+
+    //Load selected issue
+    if (this.issueId != null) {
+      //Get issue details
+      this.getIssue();
+      //Get assigned users
+      this.getAssignedUsers();
+      //Get predecessors
+      this.getAllPredecessors();
+      //get documents
+      this.getAllDocuments();
+    }
   }
 
-  //ComID 605c95b3346214a9113c549c
-  //P ID 605b80dee61730565bfe4b79
+  //Basic issue model
+  issue: Issue = {
+    createdAt: undefined,
+    state: {
+      id: '',
+      name: 'Überprüfung',
+      phase: '',
+      userGenerated: false,
+    },
+    issueDetail: {
+      name: '',
+      type: '',
+      startDate: undefined,
+      endDate: undefined,
+      expectedTime: 0,
+      progress: 0,
+      description: '',
+      requirementsAccepted: true,
+      requirementsNeeded: true,
+      requirements: [],
+      priority: 0,
+      visibility: true,
+      relevantDocuments: [],
+    },
+  };
+
+  //Load existing issue
+  getIssue() {
+    this.issueDetailsService
+      .getIssue(this.projectId, this.issueId)
+      .subscribe((data) => {
+        this.issue = data;
+        if (this.issue.issueDetail.visibility) {
+          this.visibleInput = 'extern';
+        } else {
+          this.visibleInput = 'intern';
+        }
+
+        this.stateActive = false;
+
+        (error) => {
+          console.error(error);
+        };
+      });
+  }
 
+  //Save issue
   submitForm() {
-    if (this.visibleInput === 'extern') {
-      this.visible = true;
-    } else {
-      this.visible = false;
-    }
+    if (
+      this.issue.issueDetail.name.length > 0 &&
+      this.issue.state.name.length > 0 &&
+      this.issue.issueDetail.type.length > 0 &&
+      this.visibleInput.length > 0
+    ) {
+      //Set visibilty state
+      if (this.visibleInput === 'extern') {
+        this.issue.issueDetail.visibility = true;
+      } else {
+        this.issue.issueDetail.visibility = false;
+      }
```

Comment:
Könnte man abkürzen:
```suggestion
      this.issue.issueDetail.visibility = (this.visibleInput === 'extern');
```